packages/common/src/tokens.ts

```diff
@@ -209,10 +209,13 @@ export namespace Conda {
      */
     packageChanged: ISignal<IPackageManager, Conda.IPackageChange>;
     /**
-     * Callback triggered when the user scrolls to the bottom of the package list.
-     * Optionally loads more packages, if the result is paginated.
+     * Callback triggered when the user scrolls to the bottom of the package
+     * list. Note this is a stateful method that fetches the next page of
+     * results each time it is called.
      */
-    loadMorePackages?: (environment?: string) => Promise<Array<Conda.IPackage>>;
+    loadInstalledPackages?: (
```

Comment:
Small rename and let's make it mandatory - the list of package on classical conda is getting so big, it will be good to offer the same pattern (aka searching explicitly for a package).

```suggestion
    getInstalledPackages: (
```

See https://github.com/mamba-org/gator/blob/master/packages/common/src/services.ts#L455 to figure out how to get the list for the local model.

I would propose also to add a new attribute `hasExplicitSearch` to be true if the provider is requesting explicitly a search term to list the matching package.

That will avoid the poor code casting to a boolean if `loadInstalledPackages` is undefined. And that will clarify the intent.